kernel/src/grant.rs

```diff
@@ -273,6 +282,15 @@ impl<'a, T: Default> Iterator for Iter<'a, T> {
                 return res;
             }
         }
+        // After running through all real apps, pass the debug app (idx 255) to
+        // the grant iterator in case it has state the capsule needs to process.
+        if self.index == self.len {
```

Comment:
This if statement is not needed?! (self.index == self.len must true be to pass the while loop)

Reply:
`next` is called repeatedly while iterating, having this if check (and the increment of the index variable afterwards) _is_ necessary to stop it from checking at each iteration

Reply:
Yes.